shared-bindings/audiocore/RawSample.c

```diff
@@ -27,34 +32,59 @@
 //|         :param ~circuitpython_typing.ReadableBuffer buffer: A buffer with samples
 //|         :param int channel_count: The number of channels in the buffer
 //|         :param int sample_rate: The desired playback sample rate
+//|         :param bool single_buffer: Selects single buffered or double buffered transfer mode.  This affects sample looping
+//|                                    and what happens if the sample buffer is changed while the sample is playing.
+//|                                    In single buffered transfers, samples will play once unless the play method is invoked with
+//|                                    loop=True.  A change in buffer contents will not affect active playback.
+//|                                    In double buffered transfers, samples are always looped, and changed buffer contents will
+//|                                    be played back as soon as transfer reaches the next half-buffer point.
```

Comment:
Why have loop be implicit? Why isn't it just played once as two half buffers?

Reply:
It could be done either way.  Other double-buffered constructs like `audiomixer.Mixer` and `synthio.Synthesizer` implicitly loop.  I can't think of any reason you'd want to double-buffer and not loop, but we could require the `loop=True` if that would be helpful for clarity.

Reply:
Ya, I think I'd require it. That way behavior is consistent with single and double buffered raw samples.